shopfloor_reception/services/reception.py

```diff
@@ -0,0 +1,1155 @@
+                [("id", "child_of", dest_location.id)]
+            )
+            if location not in child_locations:
+                # Scanned location isn't a child of the move's dest location
+                message = self.msg_store.dest_location_not_allowed()
+                return self._response_for_set_quantity(
+                    picking, selected_line, message=message
+                )
+            # process without pack, set destination location, and go back to
+            # `select_line`
+            selected_line.location_dest_id = location
+            return self._response_for_select_line(picking)
+
+    def _use_handlers(self, handlers, *args, **kwargs):
+        for handler in handlers:
+            response = handler(*args, **kwargs)
+            if response:
+                return response
+
+    def _assign_user_to_picking(self, picking):
+        picking.user_id = self.env.user
+
+    # DATA METHODS
+
+    def _data_for_stock_picking(self, picking, with_lines=False):
+        data = self.data.picking(picking, with_progress=True)
+        if with_lines:
+            data.update(
+                {"move_lines": self._data_for_move_lines(picking.move_line_ids)}
+            )
+        return data
+
+    def _data_for_stock_pickings(self, pickings, with_lines=False):
+        return [
+            self._data_for_stock_picking(picking, with_lines=with_lines)
+            for picking in pickings
+        ]
+
+    def _data_for_move_lines(self, lines, **kw):
+        return self.data.move_lines(lines, **kw)
+
+    # RESPONSES
+
+    def _response_for_select_document(self, pickings=None, message=None):
+        if not pickings:
+            pickings = self.env["stock.picking"].search(
+                self._domain_stock_picking(),
+                order=self._order_stock_picking(),
+            )
+        else:
+            # We sort by scheduled date first. However, there might be a case
+            # where two pickings have the exact same scheduled date.
+            # In that case, we sort by id.
+            pickings = pickings.sorted(
+                lambda p: (p.scheduled_date, p.id), reverse=False
+            )
+        data = {"pickings": self._data_for_stock_pickings(pickings, with_lines=False)}
+        return self._response(next_state="select_document", data=data, message=message)
+
+    def _response_for_set_lot(self, picking, line, message=None):
+        return self._response(
+            next_state="set_lot",
+            data={
+                "selected_move_line": self._data_for_move_lines(line),
+                "picking": self.data.picking(picking),
+            },
+            message=message,
+        )
+
+    def _response_for_set_quantity(self, picking, line, message=None):
+        return self._response(
+            next_state="set_quantity",
+            data={
+                "selected_move_line": self._data_for_move_lines(line),
```

Comment:
The line quantity to do to send to the front is the corresponding move `product_uom_qty` - `quantity_done`